### ISSUE TITLE
Hatched Turtle Home Beach Coordinates Incorrect

### ISSUE THREAD

**Issue by mrgrim** [jellysquid3/Lithium#20]
We've been experiencing an issue where newly hatched turtles do not have their home beach coordinates set correctly. Sometimes they can be off by hundreds or even thousands of blocks and can be on any block type. We've narrowed down the issue to Lithium and have reproduced it on 0.4.4 and 0.4.5.

Additionally, we've observed hatched turtles home positions changing over time seemingly randomly.

Thanks!

**Comment by DracoShadow** [comment]
Here's some screencaps of the /data get entity, the one spawned by the game doesn't alter it's HomePos coords, but the Hatched "Test" one does repeatedly. 
![2020-03-15_16 15 08](https://user-images.githubusercontent.com/57167950/76712796-f9509b80-66d8-11ea-857f-96fcba56793a.png)
![2020-03-15_16 20 26](https://user-images.githubusercontent.com/57167950/76712798-fe154f80-66d8-11ea-8739-850f425cee57.png)
![2020-03-15_16 18 24](https://user-images.githubusercontent.com/57167950/76712810-1c7b4b00-66d9-11ea-9656-886ee8a124ab.png)
![2020-03-15_16 21 51](https://user-images.githubusercontent.com/57167950/76712813-200ed200-66d9-11ea-8fbe-3b7e70dec75b.png)

**Comment by jellysquid3** [comment]
Is this occurring with the Forge or Fabric releases?

**Comment by mrgrim** [comment]
We've only tested on Fabric. I'm unsure if it's also happening with the Forge release.

**Comment by mrgrim** [comment]
Come to think of it, we tried vanilla but did not try a no mod fabric server. We can attempt that tomorrow most likely if you're having trouble reproducing.

**Comment by jellysquid3** [comment]
The cause of the issue here is that we're passing mutable block position references to the methods responsible for ticking blocks, which will cause a wide range of trouble for any code that doesn't try to convert it to an immutable reference.

In the case with turtle eggs, their random tick handler passes this mutable reference to the newly spawned turtle's home position. Since the reference is a scratch position for block ticking, the value it represents is essentially always changing and random. It can represent the position of any currently ticking block in the world!

I'm promoting this to critical status because this has the ability to cause many other problems in the game.

**Comment by mrgrim** [comment]
Oh wow, thanks for the quick turn around! Is there anything we can do to work around the issue in the meantime? Perhaps one of the config options?

**Comment by jellysquid3** [comment]
I'm currently working on backporting and wrapping up a few other bug fixes for Lithium 0.4.6. It should be available very shortly. For the meantime, you can add the following to your `lithium.toml` configuration file.
```toml
[general]
reduce_object_allocations = false # https://github.com/jellysquid3/Lithium/issues/20
```

**Comment by jellysquid3** [comment]
Lithium 0.4.6 has been released on CurseForge with a fix for this issue.
